### PR TITLE
Implement /showothers 2 (fixes #2394)

### LINKED ISSUE #2394: Show only team members in command like /showothers
Use a command like /showothers but show only teammates (in solo parts), not all.
Also, this command could be expanded to "team mode" when you see only ur mate (even in freeze scaled by alpha) and only team messages + some server messages (e.g., when you finish and get the message about how your time is better or worse)

### PR DESCRIPTION
**Pull request by def-**
in order to only show your own team members when they are in solo, not
other teams.

Also cl_show_others 2, sv_show_others_default 2

### CODE REVIEW

File: src/game/server/gamecontext.cpp [hunk 1]
```diff
@@ -2032,7 +2031,7 @@ void CGameContext::OnMessage(int MsgID, CUnpacker *pUnpacker, int ClientID)
 			if(g_Config.m_SvShowOthers && !g_Config.m_SvShowOthersDefault)
 			{
 				CNetMsg_Cl_ShowOthers *pMsg = (CNetMsg_Cl_ShowOthers *)pRawMsg;
-				pPlayer->m_ShowOthers = (bool)pMsg->m_Show;
+				pPlayer->m_ShowOthers = pMsg->m_Show;
```
Comment (heinrich5991) [inline src/game/server/gamecontext.cpp:2034]: While skimming the patch: The message type still has a boolean here. Maybe take that chance and make it an extended message.

Comment (def-) [reply src/game/server/gamecontext.cpp:2034]: Done.

File: datasrc/network.py [hunk 2]
```diff
@@ -417,6 +417,7 @@
 
 	NetMessage("Sv_TeamsState", []),
 
+	# deprecated, use showothers@netobj.ddnet.tw instead
```
Comment (heinrich5991) [inline datasrc/network.py:420]:
```suggestion
	# deprecated, use showothers2@netmsg.ddnet.tw instead
```

File: datasrc/network.py [hunk 3]
```diff
@@ -425,4 +426,8 @@
 	NetMessageEx("Sv_MyOwnMessage", "my-own-message@heinrich5991.de", [
 		NetIntAny("m_Test"),
 	]),
+
+	NetMessageEx("Cl_ShowOthers2", "showothers2@netmsg.ddnet.tw", [
```
Comment (heinrich5991) [inline datasrc/network.py:430]: Perhaps we can rename the old message to `Cl_ShowOthersLegacy` instead so we don't have to carry around the version stuff with the new message?

File: datasrc/network.py (hunk 2 above)
Comment (def-) [reply datasrc/network.py:420]: Done.

File: datasrc/network.py (hunk 3 above)
Comment (def-) [reply datasrc/network.py:430]: Done.

File: datasrc/network.py (hunk 2 above)
Comment (heinrich5991) [reply datasrc/network.py:420]:
```suggestion
	# deprecated, use showothers@netmsg.ddnet.tw instead
```

Still the wrong name after the @.

File: src/game/client/components/menus_demo.cpp [hunk 4]
```diff
@@ -927,14 +927,14 @@ void CMenus::RenderDemoList(CUIRect MainView)
 		unsigned Size = (m_lDemos[m_DemolistSelectedIndex].m_Info.m_aMapSize[0]<<24) | (m_lDemos[m_DemolistSelectedIndex].m_Info.m_aMapSize[1]<<16) |
 					(m_lDemos[m_DemolistSelectedIndex].m_Info.m_aMapSize[2]<<8) | (m_lDemos[m_DemolistSelectedIndex].m_Info.m_aMapSize[3]);
 		if(Size > 1024*1024)
-			str_format(aBuf, sizeof(aBuf), Localize("%.2f MiB"), float(Size)/(1024*1024));
+			str_format(aBuf, sizeof(aBuf), "%.2f MiB", float(Size)/(1024*1024));
 		else
-			str_format(aBuf, sizeof(aBuf), Localize("%.2f KiB"), float(Size)/1024);
+			str_format(aBuf, sizeof(aBuf), "%.2f KiB", float(Size)/1024);
```
Comment (heinrich5991) [inline src/game/client/components/menus_demo.cpp:932]: Seems like I missed this the last time, at least French uses a different name for KB: Ko.

Comment (def-) [reply src/game/client/components/menus_demo.cpp:932]: Note that this is not a Kilobyte, but a Kibibyte. But I looked it up aaaaand apparently they use Kio fo that. Huge WTF from me, will change.

**Pull request merged**